### PR TITLE
Schema validations

### PR DESCRIPTION
**Pull request by bzsurbhi**
#### Description of changes
Added schema validation for 
- manyToMany directive should have a relation name
- relationship name should not be inverse of relationName
- hasMany must only be used with with lists
- hasOne must not be used with lists
- object must be annotated with @model directive

##### CDK / CloudFormation Parameters Changed
NA

#### Description of how you validated changes
- added tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-schema-validator/src/validators/hasMany-must-be-used-with-lists.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ) as ObjectTypeDefinitionNode[];
+  objectTypeDefinitions.forEach((objectTypeDefinition) => {
+    const directiveFields = objectTypeDefinition.fields?.filter((objectField) =>
+      objectField.directives?.find((directive) => directive.name.value === 'hasMany'),
+    );
+
+    directiveFields?.forEach((directiveField) => {
+      const listType = isListType(directiveField.type);
+      if (!listType) {
+        errors.push(new InvalidDirectiveError('@hasMany must be used with a list. Use @hasOne for non-list types.'));
```
Comment (phani-srikar) [inline packages/amplify-schema-validator/src/validators/hasMany-must-be-used-with-lists.ts:25]: Is this error something we surface to the customers? Adding which field and model is being referred to in the error message would make it easier to debug.

File: packages/amplify-schema-validator/src/validators/hasOne-cannot-be-used-with-lists.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ) as ObjectTypeDefinitionNode[];
+  objectTypeDefinitions.forEach((objectTypeDefinition) => {
+    const directiveFields = objectTypeDefinition.fields?.filter((objectField) =>
+      objectField.directives?.find((directive) => directive.name.value === 'hasOne'),
+    );
+
+    directiveFields?.forEach((directiveField) => {
+      const listType = isListType(directiveField.type);
+      if (listType) {
+        errors.push(new InvalidDirectiveError('@hasOne cannot be used with lists. Use @hasMany instead'));
```
Comment (phani-srikar) [inline packages/amplify-schema-validator/src/validators/hasOne-cannot-be-used-with-lists.ts:25]: same as above. Adding which field and model is being referred to in the error message would make it easier to debug.

Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/hasOne-cannot-be-used-with-lists.ts]: fixed

File: packages/amplify-schema-validator/src/validators/hasMany-must-be-used-with-lists.ts (hunk 1 above)
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/hasMany-must-be-used-with-lists.ts]: fixed

**Review by phani-srikar — APPROVED** [review]

**Review by dpilch — APPROVED** [review]

**Pull request merged**